dockers/skylab/emptydrops/emptyDropsWrapper/emptyDropsWrapper.R

```diff
@@ -198,9 +198,29 @@ tryCatch({
                                     alpha=ed_param_alpha,
                                     BPPARAM=ed_param_BPPARAM)
 },error=function(e) {
-    cat('Error: an error occured while running emptyDrops!\n',file=stderr())
-    cat('Error: ', e$message,'\n',file=stderr())
-    quit(save="no",status=1)
+    if(grepl("need at least four unique 'x' values",e$message,fixed=TRUE))
+    {
+        # Write an empty_drops_results.csv file that has only NAs
+        print(dim(inputMatrix))
+        print(colnames(inputMatrix)[1])
+        n_rows = dim(inputMatrix)[2] # Get number of rows
+        emptyDrops_result <- matrix(data=NA,nrow=n_rows,ncol=7)
+        ## Convert output from DataFrame to data.frame
+        emptyDrops_result <- as.data.frame(emptyDrops_result)
+        colnames(emptyDrops_result) <-  c("CellId","Total", "LogProb", "PValue", "Limited", "FDR", "IsCell")
+        emptyDrops_result[,"CellId"] = colnames(inputMatrix)
+        ## Write the output file
+        catv('Writing output CSV...')
+        write.csv(x=emptyDrops_result, file=output_csv,row.names=FALSE)
+        catv('done\n')
+        quit(save="no",status=17)
```

Comment:
You will probably need to catch this error in the command block in wdl, since a non-zero return code is usually interpreted as an error

Reply:
I changed it to status 0 so the wdl can transfer the output NA file from the vm to the next task.